lib/pavilion/result_parsers.py

```diff
@@ -93,62 +95,67 @@ class ResultParserError(RuntimeError):
 
 class ResultParser(IPlugin.IPlugin):
     """Base class for creating a result parser plugin. These are essentially
-    a callable that implements operations on the test or test files. The
-    arguments for the callable are provided automatically via the test
-    config. The doc string of result parser classes is used as the user help
-    text for that class, plus the help on the config items."""
+a callable that implements operations on the test or test files. The
+arguments for the callable are provided automatically via the test
+config. The doc string of result parser classes is used as the user help
+text for that class, along with the help from the config items."""
 
     PRIO_CORE = 0
     PRIO_COMMON = 10
     PRIO_USER = 20
 
     def __init__(self, name, description, open_mode='r', priority=PRIO_COMMON):
         """Initialize the plugin object
-        :param str name: The name of this plugin.
-        :param str description: A short description of this result parser.
-        :param open_mode: How to open each file handed to the parser.
-        :param int priority: The priority of this plugin, compared to plugins
-            of the same name. Higher priority plugins will supersede others.
-        """
+
+:param str name: The name of this plugin.
+:param str description: A short description of this result parser.
+:param open_mode: How to open each file handed to the parser.
+:param int priority: The priority of this plugin, compared to plugins
+    of the same name. Higher priority plugins will supersede others.
+"""
 
         self.name = name
+        self.description = description
         self.priority = priority
         self.open_mode = open_mode
 
         super().__init__()
 
     def __call__(self, test, file, **kwargs):
         """This is where the result parser is actually implemented.
-        :param pavilion.pav_test.PavTest test: The test run object.
-        :param file: This will be a file object or a string, depending on the
-        parser's 'open_mode' setting in __init__.
-        :param dict kwargs: The arguments are the config values from the the
-        test's result config section for this parser. These should be
-        explicitly defined in your result parser class.
-        :raises ResultParserError: When something goes wrong.
-        """
+
+:param pavilion.test_run.TestRun test: The test run object.
+:param file: This will be a file object or a string, depending on the
```

Comment:
Just so you know (if you ever see this) I can't list the type of file-like objects because the library that defines them for these purposes isn't in the stdlib until 3.5. Putting it in now would cause the unittests to fail.